apis_core/generic/serializers.py

```diff
@@ -87,3 +95,100 @@ def get_content_type_key(self, obj) -> str:
     def get_content_type_name(self, obj) -> str:
         content_type = ContentType.objects.get_for_model(obj)
         return content_type.name
+
+
+class GenericModelCidocSerializer(BaseSerializer):
+    def __init__(self, *args, **kwargs):
+        self.base_uri = f"{apis_base_uri()}"
+        self.rdf_nsp_base = rdf_namespace_prefix()
+        self.appellation_nsp_prefix = f"{self.rdf_nsp_base}-appellation"
+        self.attr_nsp_prefix = f"{self.rdf_nsp_base}-attr"
+        super().__init__(*args, **kwargs)
+
+    def create_sameas(self, g, instance):
+        CRM = Namespace("http://www.cidoc-crm.org/cidoc-crm/")
+
+        # add the ID as APIS Identifier
+        apis_id = URIRef(self.attributes_namespace[f"apis-identifier/{instance.pk}"])
+        g.add((apis_id, RDF.type, CRM["E42_Identifier"]))
+        g.add((apis_id, RDFS.label, Literal(instance.pk)))
+
+        # APIS internal identifier type
+        apis_id_type = URIRef(self.attributes_namespace["apis-identifier/type"])
+        g.add((apis_id, CRM["P2_has_type"], apis_id_type))
+        g.add((apis_id_type, RDF.type, CRM["E55_Type"]))
+        g.add((apis_id_type, RDFS.label, Literal("APIS internal identifier")))
+        g.add((self.instance_uri, CRM.P1_is_identified_by, apis_id))
+
+        for uri in instance.uri_set():
+            g.add((self.instance_uri, OWL.sameAs, URIRef(uri.uri)))
+
+            for x in get_rules():
+                if m := re.match(x["match"], uri.uri):
+                    id_type = URIRef(
+                        self.attributes_namespace[x["name"] + "-identifier/type"]
+                    )
+                    g.add((id_type, RDF.type, CRM["E55_Type"]))
+                    g.add((id_type, RDFS.label, Literal(x["name"] + " ID")))
+                    id_uri = URIRef(
+                        self.attributes_namespace[
+                            x["name"] + f"-identifier/{instance.pk}"
+                        ]
+                    )
+                    g.add((id_uri, RDF.type, CRM["E42_Identifier"]))
+                    g.add((id_uri, CRM["P2_has_type"], id_type))
+                    g.add(
+                        (
+                            self.instance_uri,
+                            CRM["P1_is_identified_by"],
+                            id_uri,
+                        )
+                    )
+                    g.add((id_uri, RDFS.label, Literal(m[1])))
+
+        return g
+
+    def to_representation(self, instance):
+        g = Graph()
+        content_type = ContentType.objects.get_for_model(instance)
+
+        CRM = Namespace("http://www.cidoc-crm.org/cidoc-crm/")
+        g.namespace_manager.bind("crm", CRM, replace=True)
+        g.namespace_manager.bind("owl", OWL, replace=True)
```

Comment:
I think all the NS declarations, bindings and creation of the graph could be done in a dedicated function in the base class (maybe including the import of the custom namespaces such as CRM from a central location as suggested in another comment). That function could be called in init and store NS and g in an attribute. Like that CRM wouldnt be created in several places and all the inheriting serializers could already use it.